### ISSUE TITLE
nix-build no longer works as of latest release 2.19/2.19.1

### ISSUE THREAD

**Issue by mbwmbw1337** [kadena-io/chainweb-node#1668]
Across all OS nix-build stopped working on source as of release tag 2.19/2.19.1

**Ubuntu**
```
building '/nix/store/xmrk1v5c2jzvxlp122zg7cm5a0s87n4p-cabal2nix-chainweb.drv'...
*** cannot parse "/nix/store/vsy816yk8nfsx3l37gn68clv161f92rq-source/chainweb.cabal":
/nix/store/vsy816yk8nfsx3l37gn68clv161f92rq-source/chainweb.cabal:0:0: Unsupported cabal-version 3.8. See https://github.com/haskell/cabal/issues/4899.

*** parsing cabal file: user error (*** Exiting.)
cabal2nix: nix-prefetch-url: createProcess: posix_spawnp: does not exist (No such file or directory)
error: builder for '/nix/store/xmrk1v5c2jzvxlp122zg7cm5a0s87n4p-cabal2nix-chainweb.drv' failed with exit code 1
(use '--show-trace' to show detailed location information)
```

**MacOS**
```
building '/nix/store/z98bq03v8g0piqwv9i0zg1m8c8vdkchv-cabal2nix-chainweb.drv'...
*** cannot parse "/nix/store/vsy816yk8nfsx3l37gn68clv161f92rq-source/chainweb.cabal":
/nix/store/vsy816yk8nfsx3l37gn68clv161f92rq-source/chainweb.cabal:0:0: Unsupported cabal-version 3.8. See https://github.com/haskell/cabal/issues/4899.

*** parsing cabal file: user error (*** Exiting.)
cabal2nix: nix-prefetch-url: createProcess: posix_spawnp: illegal operation (Inappropriate ioctl for device)
error:
       … in the left operand of the update (//) operator

         at /nix/store/di36mqc6y19ivaa4qjrb2l82c6dqg7m3-source/pkgs/development/haskell-modules/lib/compose.nix:40:7:

           39|     mkDerivation = drv: (args.mkDerivation drv).override f;
           40|   })) // {
             |       ^
           41|     overrideScope = scope: overrideCabal f (drv.overrideScope scope);

       … in the left operand of the update (//) operator

         at /nix/store/di36mqc6y19ivaa4qjrb2l82c6dqg7m3-source/pkgs/development/haskell-modules/lib/compose.nix:40:7:

           39|     mkDerivation = drv: (args.mkDerivation drv).override f;
           40|   })) // {
             |       ^
           41|     overrideScope = scope: overrideCabal f (drv.overrideScope scope);

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: builder for '/nix/store/z98bq03v8g0piqwv9i0zg1m8c8vdkchv-cabal2nix-chainweb.drv' failed with exit code 1;
       last 5 log lines:
       > *** cannot parse "/nix/store/vsy816yk8nfsx3l37gn68clv161f92rq-source/chainweb.cabal":
       > /nix/store/vsy816yk8nfsx3l37gn68clv161f92rq-source/chainweb.cabal:0:0: Unsupported cabal-version 3.8. See https://github.com/haskell/cabal/issues/4899.
       >
       > *** parsing cabal file: user error (*** Exiting.)
       > cabal2nix: nix-prefetch-url: createProcess: posix_spawnp: illegal operation (Inappropriate ioctl for device)
       For full logs, run 'nix-store -l /nix/store/z98bq03v8g0piqwv9i0zg1m8c8vdkchv-cabal2nix-chainweb.drv'.
```

**Comment by enobayram** [comment]
@mbwmbw1337 Thank you for opening this issue. We have #1646 to fix the MacOS build issue, but it got blocked behind a minor internal discussion point regarding the CI, I'll make sure to push it to completion, at least the part that fixes the `nix build` itself.

That said, I'm completely surprised by your Ubuntu build error. I have never encountered that type of error neither locally nor in the CI. What version of Ubuntu and Nix are you using? Would you mind providing the whole command you use to invoke the Nix build? Also, do you have any changes in your local checkout of the `chainweb-node` repo? 

You've indicated under #1667 that you're getting the same error, but that PR fixes the Linux builds in the CI.

**Comment by enobayram** [comment]
@mbwmbw1337 seems like you're using `nix-build` and not `nix build`, which means you're using the old pre-flake nix definitions. We will soon push a PR to bring `nix-build` in line with the new nix-flake-based build infra, but for now, you should be able to run  `nix build` or `nix-build default-flake.nix`.

We can close this issue once plain `nix-build` is working again.

**Comment by enobayram** [comment]
@mbwmbw1337 After merging #1646 and #1670, this issue should be completely fixed now. Let me know if you're still getting errors, otherwise I'll close this issue.

**Comment by mbwmbw1337** [comment]
I have confirmed this issue is fixed by #1646 and #1670! Thank you again @enobayram for your excellent and quick support.

**Comment by enobayram** [comment]
@mbwmbw1337 Thank you for confirming, glad it worked for you!
